Tests/HummingbirdTests/ApplicationTests.swift

```diff
@@ -531,6 +531,22 @@ final class ApplicationTests: XCTestCase {
         XCTAssertEqual(MyService.shutdown.load(ordering: .relaxed), true)
     }
 
+    func testRunBeforeServer() async throws {
+        let runBeforeServer = ManagedAtomic(false)
+        let router = HBRouter()
+        var app = HBApplication(
+            responder: router.buildResponder()
+        )
```

Comment:
Once #339 is merged I can add a check to see if the atomic has been set by then

Reply:
#339 can be merged